src/components/common/SelectorButtonContainer/index.tsx

```diff
@@ -0,0 +1,114 @@
+import styled from '@emotion/styled'
+import { useState } from 'react'
+
+import { palette } from '@/styles/palette'
+
+import AlertText from '../AlertText'
+import CountNumber from '../CountNumber'
+import CustomSelectorButton from '../CustomSelectorButton'
+import SelectorButton from '../SelectorButton'
+
+type SelectorButtonContainerProps = {
+  isDarkMode: boolean
+  buttonNames: string[]
+  maxLength: number
+}
+
+const SelectorButtonContainer = ({
+  isDarkMode,
+  buttonNames,
+  maxLength,
+}: SelectorButtonContainerProps) => {
+  const [selectedCount, setSelectedCount] = useState(0)
+  const [showAlert, setShowAlert] = useState(false)
+  const [customButtons, setCustomButtons] = useState<string[]>([])
+
+  const handleButtonSelection = (isSelected: boolean) => {
+    if (isSelected) {
+      if (selectedCount + 1 > maxLength) {
+        setShowAlert(true)
+        return
+      } else {
+        setSelectedCount(selectedCount + 1)
+      }
+    } else {
+      setSelectedCount(Math.max(selectedCount - 1, 0))
+    }
```

Comment:
p3: 분기 처리가 많아서. 시간이 지나서 코드를 읽을 때 한번에 이해가되지 않을 것 같습니다. 아래처럼 바꾸면 어떨까요?
```suggestion
  const handleButtonSelection = (isSelected: boolean) => {
    if (!isSelected) {
        const newSelectedCount = Math.max(selectedCount - 1, 0);
        setSelectedCount(newSelectedCount);
        return;
    }

    const isOverMaxLength = selectedCount + 1 > maxLength;
    if (isOverMaxLength) {
        setShowAlert(true);
        return;
    }

    setSelectedCount(selectedCount + 1);
}
```